content/_ext/yaml_gallery_generator.py

```diff
@@ -2,17 +2,95 @@
 from textwrap import dedent
 import pathlib
 
-def build_from_yaml(filename, display_name):
 
-    with open(f'{filename}.yaml') as fid:
-        items = yaml.safe_load(fid)
+def _tag_in_item(item, tag_str):
```

Comment:
With the check on `tag_str` being `None`, should `tag_str` be a keyword argument with the default value being `None`?

If so, accept this suggestion:

```suggestion
def _tag_in_item(item, tag_str=None):
```